### PR TITLE
bump heapless and typenum

### PR DESCRIPTION
**Pull request by tarfu**

### CODE REVIEW

**Comment by tarfu** [comment]
Are there any issues with the bump in your eyes @nickray ?

File: Cargo.toml [hunk 1]
```diff
@@ -8,13 +8,13 @@ categories = ["embedded", "encoding", "no-std"]
 keywords = ["serde", "cbor", "rust", "no-std"]
 repository = "https://github.com/ycrypto/heapless-bytes"
 readme = "README.md"
-edition = "2018"
+edition = "2021"
 
 [dependencies]
-typenum = "1.11.2"
+typenum = ">=1.11.2"
```
Comment (sosthene-nitrokey) [inline Cargo.toml]: What does this change?

File: Cargo.toml [hunk 2]
```diff
@@ -8,13 +8,13 @@ categories = ["embedded", "encoding", "no-std"]
...
 readme = "README.md"
-edition = "2018"
+edition = "2021"
 
 [dependencies]
 typenum = "1.11.2"
 
 [dependencies.heapless]
-version = "0.7"
+version = "^0.8"
```
Comment (sosthene-nitrokey) [inline Cargo.toml]: Why the caret?

`version = "0.8"` does the same: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements

**Comment by tarfu** [comment]
Closing this PR in favor of #3 which does more and also cleans up.

**Pull request closed without merging**